request/http_retransport.go

```diff
@@ -1,41 +1,102 @@
 package request
 
 import (
+	"io"
+	"log"
 	"net/http"
 	"sync"
 
 	"github.com/NordSecurity/nordvpn-linux/events"
+	"github.com/NordSecurity/nordvpn-linux/internal"
 )
 
-// HTTPReTransport is std RoundTripper enhanced to reconnect after vpn connect is done
+// HTTPReTransport is modified/enhanced RoundTripper.
 // Thread safe.
 type HTTPReTransport struct {
-	inner    http.RoundTripper
-	createFn func() http.RoundTripper
-	mu       sync.RWMutex
+	// for protects the access to inner and to shouldRecreate. This will allow multiple
+	// requests in the same time for inner, but only one can recreate it
+	mu sync.RWMutex
+	// when executing the requests set this to true on error and when inner is recreated is set
+	// to false. This is used to ensure only one failed request will recreate inner value on
+	// error.
+	inner           http.RoundTripper
+	createFn        RoundTripperCreateFunc
+	protoMajor      int
+	protoMinor      int
+	proto           string
+	shouldRetryFunc ShouldRetryFunc
 }
 
-func NewHTTPReTransport(fn func() http.RoundTripper) *HTTPReTransport {
+// RoundTripperCreateFunc is a function used to create a new instance of round tripper when it needs
+// to be recreated.
+type RoundTripperCreateFunc func() http.RoundTripper
+
+// ShouldRetryFunc defines a function that determines whether an HTTP request should be retried
+// based on the given error.
+type ShouldRetryFunc func(err error) bool
+
+// NewHTTPReTransport is a default constructor function for a HTTPReTransport.
```

Comment:
absolutely petty nitpick:
```suggestion
// NewHTTPReTransport is a default constructor function for an HTTPReTransport.
```